docs/design/sql/05-client-support.md

```diff
@@ -87,38 +91,33 @@ response {
 The `fetch` command extracts the next page of rows from the server-side cursor via query ID. 
 
 Data fetching is organized in a simple request-response fashion: one page is returned for every `fetch` request. This 
-implementation is not optimal for large result sets due to increased latency. A better implementation will stream data from the 
-member to the client without waiting for an explicit `fetch` request for every page. We decided not to do this in 4.1 due to 
-time constraints. However, the design of the client protocol allows for such interaction, so we may implement the streaming 
-protocol in the future. 
-
-The command must be sent to the query initiator for two reasons:
-1. This guarantees the optimal performance, because the data is extract with a minimal number of network requests 
-1. Currently, there is no way to re-route `fetch` requests from the one member to another. This limitation might be relaxed in 
-the future.
+implementation might be not optimal for large result sets due to increased latency. An alternative implementation could stream
+data from the member to the client without waiting for an explicit `fetch` request for every page. We decided not to do this in
+4.1 due to time constraints. However, the design of the client protocol allows for such interaction, so we may implement the 
+streaming protocol in the future. 
 
 ```
 request {
     queryId : SqlQueryId
+    cursorBufferSize : int
 }
 
 response {
-    rowPage : List<SqlRow>
-    rowPageLast : boolean
+    rowPage : SqlPage
     error : SqlError
 }
 ```
 
-### 2.3 Close Command
-
-The `close` command stops query execution on the member, and releases the associated resources. 
+The `fetch` command must be sent to the query initiator only. There are two reasons for this:
+1. This guarantees the optimal performance, because the data is extracted with a minimal number of network requests
+1. Currently, there is no way to re-route `fetch` requests from the one member to another. This limitation might be relaxed in
+   the future.
 
-If the previous `execute` or `fetch` response returned "end of data" flag (`rowPageLast == true`), then it is not necessary 
-to send this command to the member.
-
-If the server-side cursor with the given ID is not found, the command is no-op.
+### 2.3 Close Command
 
-The command must be sent to the query initiator for reasons similar to the `fetch` command.
+The `close` command stops query execution on the member, and releases the associated resources. The request could be sent
+without waiting for the initial response from the `execute` command. If the server-side cursor with the given ID is not found, 
+the command is no-op.
```

Comment:
The last sentence is no longer true. We create a close marker because we don't know whether the `execute` command is expected or whether the query is already completed.

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/18182/commits/cc97342fbc74025d955ec9840f103c70d4769b1b